app/cli/internal/action/attestation_init.go

```diff
@@ -229,6 +229,9 @@ func (action *AttestationInit) Run(ctx context.Context, opts *AttestationInitRun
 		return "", err
 	}
 
+	// Resolve the hosted context information
+	action.c.ResolveHostedRunnerContext(ctx)
+
 	// Load the env variables both the system populated and the user predefined ones
 	if err := action.c.ResolveEnvVars(ctx, attestationID); err != nil {
```

Comment:
why instead of creating a new method `ResolveHostedRunnerContext` we make it part of `ResolveEnvVars`? renaming it of course i.e `ResolveRunnerInfo`

That way we only write in the state once